### PR TITLE
fix: PHP 8 problems

### PR DESCRIPTION
**Pull request by phanan**
Since Travis actually didn't run for a while, this library hasn't been tested against PHP 8. This PR aims to fix the issues that come with this problem.

### Todos:

- [x] Upgrade PHPUnit
- [x] Update tests to work with PHPUnit's method changes, removals, and deprecations
- [x] Upgrade PHP-CS-Fixer
	- [x] Temporarily allow PHP-CS-Fixer to fail for PHP~8, as the tool doesn't support it yet
- [x] Change the `AbstractTestCase::expectArgumentError` method to adapt to the new message format thrown in PHP 8 (actually, create a new method altogether).
- [ ] Fix the few broken tests, which might come from broken functionalities themselves. @lstrojny maybe can lend a hand here?

This PR also switches the CI from Travis, whose performance and support have been shown to be subpar to that of GitHub Actions lately, to the latter.

### CODE REVIEW

**Comment by lstrojny** [comment]
FYI: finally got around finishing the work on switching to GH actions. That should help this branch a bit already since at least here is a failing PHP 8 build.

File: tests/Functional/FunctionalTest.php [hunk 1]
```diff
@@ -24,7 +24,7 @@ public function testAllDefinedConstantsAreValidCallables()
         $functions = $functionalClass->getConstants();
 
         foreach ($functions as $function) {
-            $this->assertInternalType('callable', $function);
+            $this->assertTrue(is_callable($function));
```
Comment (lstrojny) [inline tests/Functional/FunctionalTest.php:27]: `assertIsCallable()` should work

**Review by lstrojny — COMMENTED** [review]
Great work @phanan, thank you very much. What can I do to help?

**Comment by phanan** [comment]
@lstrojny You can help with the failing tests ;) Feel free to take over and push to this branch.

**Comment by lstrojny** [comment]
#238 has landed

**Pull request closed without merging**